server/doltgres_handler.go

```diff
@@ -432,18 +454,33 @@ func (h *DoltgresHandler) maybeReleaseAllLocks(c *mysql.Conn) {
 // These nodes will eventually return an OK result, but their intermediate forms here return a different schema
 // than they will at execution time.
 func nodeReturnsOkResultSchema(node sql.Node) bool {
-	switch node.(type) {
-	case *plan.InsertInto, *plan.Update, *plan.UpdateJoin, *plan.DeleteFrom:
+	switch n := node.(type) {
+	case *plan.InsertInto:
+		if len(n.Returning) > 0 {
+			return false
+		}
+	case *plan.Update:
+		if len(n.Returning) > 0 {
+			return false
+		}
```

Comment:
Wouldn't these also have an `else` branch that returns `true`? At least that's equivalent to the old logic, but perhaps we're now supposed to do an `OkResult` check for these too.

Or it could be `return len(n.Returning) == 0`